packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,141 @@
+
+  @override
+  double childCrossAxisPosition(RenderSliver child) {
+    switch (constraints.axisDirection) {
+      case AxisDirection.up:
+      case AxisDirection.down:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dx;
+      case AxisDirection.left:
+      case AxisDirection.right:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dy;
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // First, layout each child with flex == 0.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.flex ?? 1;
+      if (flex == 0) {
+        child.layout(constraints, parentUsesSize: true);
+        usedUpExtent += child.geometry!.crossAxisExtent;
+        print(usedUpExtent);
+      } else {
+        totalFlex += flex;
+      }
+      child = childAfter(child);
+    }
+    final double extentPerFlexValue = (crossAxisExtent - usedUpExtent) / totalFlex;
+
+    child = firstChild;
+    double offset = 0.0;
+
+    // At this point, all slivers with constrained cross axis should already be laid out.
+    geometry = SliverGeometry.zero;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.flex ?? 1;
+      double childExtent;
+      if (flex != 0) {
+        childExtent = extentPerFlexValue * flex;
+        child.layout(constraints.copyWith(
+          crossAxisExtent: extentPerFlexValue * flex,
+        ), parentUsesSize: true);
+      } else {
+        childExtent = child.geometry!.crossAxisExtent;
+      }
+      // Set child parent data.
+      switch (constraints.axis) {
+        case Axis.vertical:
+          childParentData.paintOffset = Offset(offset, 0.0);
+        case Axis.horizontal:
+          childParentData.paintOffset = Offset(0.0, offset);
+      }
+      offset += childExtent;
+      if (geometry!.scrollExtent < child.geometry!.scrollExtent) {
+        geometry = child.geometry;
+      }
+      child = childAfter(child);
+    }
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    RenderSliver? child = firstChild;
+
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      context.paintChild(child, offset + childParentData.paintOffset);
+      child = childAfter(child);
+    }
+  }
+
+  @override
+  void applyPaintTransform(RenderSliver child, Matrix4 transform) {
```

Comment:
Is this tested?

Reply:
I don't think I'll test this because I just used the same implementation as ProxySliver, and none of the other slivers test this either :.

Reply:
Here is where I meant it: Then why is it being overridden here? Should this extend the RenderProxySliver class instead of RenderSliver?
It not being tested somewhere else is not a reason to not test it here.